tests/system/python/api/test_plugin_discovery.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+""" Test plugin discovery (north, south, filter, notificationDelivery, notificationRule) REST API """
+
+import subprocess
+import http.client
+import json
+from collections import Counter
+import pytest
+
+
+__author__ = "Ashish Jabble"
+__copyright__ = "Copyright (c) 2019 Dianomic Systems"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+def install_plugin(type, plugin, branch="develop", plugin_lang="python", use_pip_cache=True):
+    if plugin_lang == "python":
+        path = "$FOGLAMP_ROOT/tests/system/python/scripts/install_python_plugin {} {} {} {}".format(
+            branch, type, plugin, use_pip_cache)
+    else:
+        path = "$FOGLAMP_ROOT/tests/system/python/scripts/install_c_plugin {} {} {}".format(
+            branch, type, plugin)
+    try:
+        subprocess.run([path], shell=True, check=True)
+    except subprocess.CalledProcessError:
+        assert False, "{} plugin installation failed".format(plugin)
+
+    # Cleanup code
+    subprocess.run(["rm -rf /tmp/foglamp-{}-{}".format(type, plugin)], shell=True, check=True)
+    # TODO: FOGL-2642 - Remove notificationDelivery And notificationRule plugins
```

Comment:
let's fix 2642 first.

Reply:
fyi, We are not BLOCKED with it as we can fetch notification plugins from `curl -sX GET http://localhost:8081/foglamp/notification/plugin | jq` And GUI is using the same endpoint.

Reply:
> let's fix 2642 first.

https://github.com/foglamp/FogLAMP/pull/1486